framework/src/controller/schema/constants.js

```diff
@@ -133,330 +160,253 @@ module.exports = {
+							},
+						},
 					},
 					LIFETIME: {
-						MINIMUM: 1,
-						MAXIMUM: 72,
-					},
-					KEYSGROUP: {
-						MIN_ITEMS: 1,
-						MAX_ITEMS: 15,
+						type: 'object',
+						required: ['MINIMUM', 'MAXIMUM'],
+						properties: {
+							MINIMUM: {
+								type: 'integer',
+								min: 1,
+								description:
+									'Minimum timeframe in which a multisignature transaction will exist in memory before the transaction is confirmed',
+							},
+							MAXIMUM: {
+								type: 'integer',
+								min: 1,
+								description:
+									'Maximum timeframe in which multisignature transaction will exist in memory before the transaction is confirmed',
+							},
+						},
+						additionalProperties: false,
 					},
 				},
+				additionalProperties: false,
 			},
 			NETHASHES: {
 				type: 'array',
 				items: {
 					type: 'string',
 					format: 'hex',
 				},
-				default: [
-					// Mainnet
-					'ed14889723f24ecc54871d058d98ce91ff2f973192075c0155ba2b7b70ad2511',
-					// Testnet
-					'da3ed6a45429278bac2666961289ca17ad86595d33b31037615d4b8e8f158bba',
-				],
 				description: 'For mainnet and testnet',
 			},
 			NORMALIZER: {
 				type: 'string',
 				format: 'amount',
-				default: '100000000',
 				description: 'Use this to convert LISK amount to normal value',
 			},
 			REWARDS: {
-				$ref: 'rewards',
-				default: {
-					MILESTONES: [
-						'500000000', // Initial Reward
-						'400000000', // Milestone 1
-						'300000000', // Milestone 2
-						'200000000', // Milestone 3
-						'100000000', // Milestone 4
-					],
-					OFFSET: 2160,
-					DISTANCE: 3000000,
+				type: 'object',
+				required: ['MILESTONES', 'OFFSET', 'DISTANCE'],
+				description: 'Object representing LSK rewards milestone',
+				properties: {
+					MILESTONES: {
+						type: 'array',
+						items: {
+							type: 'string',
+							format: 'amount',
+						},
+						description: 'Initial 5, and decreasing until 1',
+					},
+					OFFSET: {
+						type: 'integer',
+						min: 1,
+						description: 'Start rewards at block (n)',
+					},
+					DISTANCE: {
+						type: 'integer',
+						min: 1,
+						description: 'Distance between each milestone',
+					},
 				},
+				additionalProperties: false,
 			},
 			TOTAL_AMOUNT: {
 				type: 'string',
 				format: 'amount',
-				default: '10000000000000000',
 				description:
 					'Total amount of LSK available in network before rewards milestone started',
 			},
 			TRANSACTION_TYPES: {
-				$ref: 'transactionTypes',
-				default: {
-					SEND: 0,
-					SIGNATURE: 1,
-					DELEGATE: 2,
-					VOTE: 3,
-					MULTI: 4,
-					DAPP: 5,
-					IN_TRANSFER: 6,
-					OUT_TRANSFER: 7,
+				type: 'object',
+				required: [
+					'SEND',
+					'SIGNATURE',
+					'DELEGATE',
+					'VOTE',
+					'MULTI',
+					'DAPP',
+					'IN_TRANSFER',
+					'OUT_TRANSFER',
+				],
+				properties: {
+					SEND: {
+						type: 'integer',
+						enum: [0],
```

Comment:
As this will only have one value, I suggest looking at `const` for JSON schema: http://json-schema.org/latest/json-schema-validation.html#rfc.section.6.1.3

`const: 0`

Reply:
Same applies to other tx types.

Reply:
Yes current version of `ajv` introduced the support of `const` keyword. We can use it here.

Reply:
@limiaspasdaniel May be this is related to changes you are doing.